File: backend/AdPlatform/DTOs/Ads/AdDto.cs
using AdPlatform.DTOs.Users;

namespace AdPlatform.DTOs.Ads;

public class AdDto
{
    public int Id { get; set; }

    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public int Price { get; set; }

    public bool IsSold { get; set; }

    public DateTimeOffset CreatedAt { get; set; }
    public DateTimeOffset? UpdatedAt { get; set; }

    public int CategoryId { get; set; }
    public string CategoryName { get; set; } = string.Empty;

    public int CityId { get; set; }
    public string CityName { get; set; } = string.Empty;

    public UserDto User { get; set; } = null!;

    public List<ImageDto> Images { get; set; } = [];
}

File: frontend/src/types/DTOs/adDto.ts
import { ImageDto } from '@/types/DTOs/imageDto.ts';
import { UserDto } from '@/types/DTOs/userDto.ts';

export interface AdDto {
  id: number;
  title: string;
  description: string;
  price: number;
  isSold: boolean;
  createdAt: string;
  updatedAt?: string;
  categoryId: string;
  categoryName: string;
  cityId: string;
  cityName: string;
  user: UserDto;
  images: ImageDto[];
  isFavorite: boolean;
}
